spec/workers/slack_notification_worker_spec.rb

```diff
@@ -0,0 +1,53 @@
+require 'rails_helper'
+
+RSpec.describe SlackNotificationWorker do
+  include TestHelpers::LoggingHelper
+  let(:rails_config) { environment_config_double }
+
+  describe 'SlackNotificationWorker' do
+    let(:text) { 'example text' }
+    let(:url) { 'https://example.com/support' }
+    let(:output) { capture_logstash_output(rails_config) { invoke_worker } }
+
+    def invoke_worker
+      SlackNotificationWorker.new.perform(text, url)
+    end
+
+    it 'logs it has run to the default Rails logger' do
+      expect(output).not_to be_blank
+    end
+
+    it 'includes the text supplied in the log' do
+      expect(output).to match(text)
+    end
+
+    it 'does not send a Slack notification if STATE_CHANGE_SLACK_URL is empty' do
+      stub_const('SlackNotificationWorker::INCOMING_WEBHOOK_URL', nil)
+      expect(HTTP).not_to receive(:post)
+      invoke_worker
+    end
+
+    context 'when STATE_CHANGE_SLACK_URL is set' do
+      before do
+        stub_const('SlackNotificationWorker::INCOMING_WEBHOOK_URL', 'https://example.com/webhook')
+      end
+
+      it 'sends a Slack notification to this webhook' do
+        expect(HTTP).to receive(:post)
```

Comment:
Let's go with Rubocop here and do a `allow()` bla `expect()` thing.

Reply:
Ok